src/file/file.model.ts

```diff
@@ -76,14 +76,39 @@ fileSchema.virtual('fullExtension')
   });
 
 // handleE11000 is called when there is a duplicateKey Error
-const handleE11000 = function (error: MongoError, _: any, next: NextFunction) {
+const handleE11000 = function (error: MongoError, _: any, next: NextFunction) : void {
   if (error.name === 'MongoError' && error.code === 11000) {
-    next(new KeyAlreadyExistsError(this.key));
+    const retMessage : string = getMongoErrorIndices(error);
+    next(new KeyAlreadyExistsError(retMessage));
```

Comment:
Maybe we should create a different error for cases where the compound indexes already exist. We can accidentally confuse it with the key field.

Reply:
Changed to 'UniqueIndexExistsError'